packages/component_code_gen/code_gen/generate_component_code.py

```diff
@@ -1,76 +1,94 @@
+        db = supabase_helpers.SupabaseConnector()
 
-    if config['enable_docs'] == False:
-        logger.warn("docs are disabled")
-        return no_docs(app, prompt, templates)
+        auth_meta = db.get_app_auth_meta(app)
+        # TODO: is this needed only for actions?
+        add_code_example(templates, auth_meta['component_code_scaffold_raw'])
 
-    docs_meta = db.get_app_docs_meta(app)
+        docs_meta = db.get_app_docs_meta(app)
+        # Initialize a flag to track if we obtained any results with docs
+        has_docs_result = False
 
-    try:
         if 'docs_url' in docs_meta:
             contents = db.get_docs_contents(app)
             if contents:
-                docs =  { row['url']: row['content'] for row in contents }
-                return with_docs(app, prompt, docs, 'api reference', templates)
-    except Exception as e:
-        logger.error(e)
-        logger.error("failed with docs")
+                docs = {row['url']: row['content'] for row in contents}
+                results.append(call_langchain(
+                    app, prompt, templates, docs, 'api reference'))
+                has_docs_result = True
 
-    try:
         if 'openapi_url' in docs_meta:
             contents = db.get_openapi_contents(app)
             if contents:
-                docs = { row['path']: row['content'] for row in contents }
-                return with_docs(app, prompt, docs, 'openapi', templates)
-    except Exception as e:
-        logger.error(e)
-        logger.error("failed with openapi")
+                docs = {row['path']: row['content'] for row in contents}
+                results.append(call_langchain(
+                    app, prompt, templates, docs, 'openapi'))
+                has_docs_result = True
+
+        # If we haven't obtained any results using docs
+        if not has_docs_result:
+            results.append(call_langchain(app, prompt, templates))
 
-    return no_docs(app, prompt, templates)
+    # Create a new prompt string
+    new_prompt = "I've asked other GPT agents to generate the following code based on the prompt and the instructions below. One set of code (or all) may not follow the rules laid out in the prompt or in the instructions below, so you'll need to review it for accuracy. Try to evaluate the examples according to the rules, combine the best parts of each example, and generate a final set of code that solves the problem posed by the prompt and follows all of the rules below. Here are the attempts + code:\n\n---\n\n"
+    for idx, result in enumerate(results, 1):
+        new_prompt += f"Try {idx}:\n\n${result}\n\n---\n\n"
+    logger.debug(f"\n\n---------------{new_prompt}\n\n")
+    new_prompt += "---\n\n" + prompt
 
+    # Call the model again with the new prompt to get the final result
+    logger.debug(f"Calling the model a final time to summarize the attempts")
+    return call_langchain(app, new_prompt, templates)
 
-def no_docs(app, prompt, templates):
```

Comment:
In this refactor I combined `no_docs` and `with_docs` into a single function — `call_langchain`